packages/openapi-generator/src/parser/operation.ts

```diff
@@ -1,50 +1,141 @@
 import { $Refs } from '@apidevtools/swagger-parser';
 import { OpenAPIV3 } from 'openapi-types';
-import { Method, OpenApiOperation } from '../openapi-types';
+import {
+  camelCase,
+  filterDuplicatesRight,
+  partition,
+  UniqueNameGenerator
+} from '@sap-cloud-sdk/util';
+import { OpenApiOperation, OpenApiParameter } from '../openapi-types';
 import { parseRequestBody } from './request-body';
-import { parseParameters } from './parameters';
+import { resolveObject } from './refs';
+import { parseSchema } from './schema';
+import { parseResponses } from './responses';
+import { OperationInfo } from './operation-info';
 
+/**
+ * Parse an operation info into a serialization-ready object.
+ * @param operationInfo Parsing relevant information on an operation.
+ * @param pathItem The path item object, the operation occurred in.
+ * @param refs List of crossreferences that can occur in the document.
+ * @returns A flat list of parsed operations.
+ */
 export function parseOperation(
-  path: string,
+  { operation, pathPattern, method }: OperationInfo,
   pathItem: OpenAPIV3.PathItemObject,
-  method: Method,
   refs: $Refs
 ): OpenApiOperation {
-  const operation = getOperation(pathItem, method);
-  // TODO: What does the OpenApi generator do in this case?
   const requestBody = parseRequestBody(operation.requestBody, refs);
-  const parameters = parseParameters(operation, refs);
+  const response = parseResponses(operation.responses, refs);
+  const relevantParameters = getRelevantParameters(
+    [...(pathItem.parameters || []), ...(operation.parameters || [])],
```

Comment:
What happens with duplicates entires? Can this happen? Is this a problem? Are the parameters the query or path parameters?

Reply:
forget the question just read the code below and the implementation.